weak_to_strong/train.py

```diff
@@ -17,6 +17,47 @@
 from weak_to_strong.loss import xent_loss
 from weak_to_strong.model import TransformerWithHead
 
+def try_initialize_model_with_kwargs(model_config, custom_kwargs, num_labels=2, linear_probe=False):
+    try:
+        # Attempt to initialize the model with the given arguments
+        return TransformerWithHead.from_pretrained(
+            model_config.name, num_labels=2, linear_probe=linear_probe, **custom_kwargs
+        ).to("cuda")
+    except TypeError as e:
+        # Check if the exception is due to an unexpected keyword argument
+        if "unexpected keyword argument" in str(e):
+            # Identify the problematic keyword
+            problem_kwarg = str(e).split("'")[1]
+
+            # Remove the problematic keyword and retry
+            new_kwargs = {k: v for k, v in custom_kwargs.items() if k != problem_kwarg}
+            return try_initialize_model_with_kwargs(model_config, new_kwargs, num_labels, linear_probe)
+        else:
+            # If the error is not due to an unexpected keyword argument, re-raise it
+            raise
+
+def try_initialize_model_with_kwargs_2(model_config, custom_kwargs, num_labels=2, linear_probe=False):
```

Comment:
the only difference is device_map="auto"?  you could just pass dict(**custom_kwargs, device_map="auto") and use a single function